sale_cutoff_time_delivery/models/stock_picking.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from dateutil.relativedelta import relativedelta
+from psycopg2 import sql
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    cutoff_time_hms = fields.Char(
+        compute="_compute_cutoff_time_HMS", store=True, default="00:00"
+    )
+    cutoff_time_diff = fields.Integer(
+        compute="_compute_cutoff_time_diff",
+        search="_search_cutoff_time_diff",
+        store=False,
+    )
+
+    @api.depends("location_id")
+    def _compute_cutoff_time_diff(self):
+        """Compute the stock.picking status in relation to warehouse cut-off time.
+
+        Possible values are:
+        -1 schedulled_date is in the past of yesterday's cutoff time
+         0 scheduled_date is between yesterday and today's cuttoff
+         1 scheduled_date is in the future of today's cutoff time
+
+        """
+        for record in self:
+            warehouse = record.location_id.get_warehouse()
+            hour, minute = warehouse._get_hour_min_from_value(warehouse.cutoff_time)
+            today_cutoff = fields.Datetime.now().replace(
+                hour=hour, minute=minute, second=0
+            )
+            yesterday_cutoff = today_cutoff + relativedelta(days=-1)
+            if record.scheduled_date < yesterday_cutoff:
+                record.cutoff_time_diff = -1
+            elif record.scheduled_date > today_cutoff:
+                record.cutoff_time_diff = 1
+            else:
+                record.cutoff_time_diff = 0
+
+    def _search_cutoff_time_diff(self, operator, value):
+        if operator not in ("=", "!="):
+            raise UserError(_("Unsupported search operator %s") % (operator,))
+        today = fields.Datetime.now()
+        yesterday = today - relativedelta(days=1)
+        params = {
+            "yesterday": yesterday.strftime("%Y-%m-%d"),
+            "today": today.strftime("%Y-%m-%d"),
```

Comment:
As the format expected by Pg is already defined in Odoo internals (https://github.com/odoo/odoo/blob/13.0/odoo/tools/misc.py#L584), you could do:
```suggestion
            "yesterday": fields.Date.to_string(yesterday),
            "today": fields.Date.to_string(today),
```

Reply:
Ok, I changed it.